tests/test_CompareEclipseRestart.cpp

```diff
@@ -0,0 +1,173 @@
+#include "test_CompareEclipseRestart.hpp"
+
+int main(int argc, char** argv) {
+    char* gridFile1 = argv[1];
+    char* gridFile2 = argv[2];
+    char* unrstFile1 = argv[3];
+    char* unrstFile2 = argv[4];
+
+    if (argc != 5) {
+        printHelp();
+    }
+    else {
+        // Comparing grid sizes from .EGRID-files
+        {
+            ecl_grid_type* ecl_grid1 = ecl_grid_alloc(gridFile1);
+            ecl_grid_type* ecl_grid2 = ecl_grid_alloc(gridFile2);
+
+            std::cout << "\nName of grid1: " << ecl_grid_get_name(ecl_grid1) << std::endl;
+            int gridCount1 = ecl_grid_get_global_size(ecl_grid1);
+            std::cout << "Grid1 count = " << gridCount1 << std::endl;
+            std::cout << "Name of grid2: " << ecl_grid_get_name(ecl_grid2) << std::endl;
+            int gridCount2 = ecl_grid_get_global_size(ecl_grid2);
+            std::cout << "Grid2 count = " << gridCount2 << std::endl;
+
+            ecl_grid_free(ecl_grid1);
+            ecl_grid_free(ecl_grid2);
+        }
+
+        //Comparing keyword values from .UNRST-files:
+
+        ReadUNRST read;
+        if (read.open(unrstFile1, unrstFile2)) {
+            for (const char* keyword : {"SGAS", "SWAT", "PRESSURE"}) {
+                std::cout << "\nKeyword " << keyword << ":\n\n";
+
+                std::vector<double> absDeviation, relDeviation;
+                read.results(keyword, &absDeviation, &relDeviation);
+
+                std::cout << "absDeviation size = " << absDeviation.size() << std::endl;
+                std::cout << "relDeviation size = " << relDeviation.size() << std::endl;
+                std::cout << "Average absolute deviation = " << ReadUNRST::average(absDeviation) << std::endl;
+                std::cout << "Median absolute deviation = " << ReadUNRST::median(absDeviation) << std::endl;
+                std::cout << "Average relative deviation = " << ReadUNRST::average(relDeviation) << std::endl;
+                std::cout << "Median relative deviation = " << ReadUNRST::median(relDeviation) << std::endl;
+            }
+            std::cout << std::endl;
+            read.close();
+        }
+    }
+    return 0;
+}
+
+//------------------------------------------------//
+
+void printHelp() {
+    std::cout << "The program takes four arguments:\n"
+        << "1. .EGRID-file number 1\n" 
+        << "2. .EGRID-file number 2\n" 
+        << "2. .UNRST-file number 1\n" 
+        << "3. .UNRST-file number 2\n";
+}
+
+bool ReadUNRST::open(const char* unrstFile1, const char* unrstFile2) {
+    if (ecl_file1 == NULL) {
+        ecl_file1 = ecl_file_open(unrstFile1, ECL_FILE_CLOSE_STREAM);
+    }
+    if (ecl_file2 == NULL) {
+        ecl_file2 = ecl_file_open(unrstFile2, ECL_FILE_CLOSE_STREAM);
+    }
+
+    if (ecl_file1 == NULL || ecl_file2 == NULL) {
+        return false;
+    }
+    else {
+        return true;
+    }
+}
+
+void ReadUNRST::close() {
+    ecl_file_close(ecl_file1);
+    ecl_file_close(ecl_file2);
+}
+
```

Comment:
It is good to actively use whitespace to separate methods/blocks/classes etc. Use more whitespace to separate higher level constructs. My rule of thumb:
 - max one blank line (in a row) inside a function
 - use more (2-5) blank lines between functions (of a class)
 - use even more (4-5) blank lines between classes

Temper with your own common sense. Many people think 5 blank lines sounds crazy, but discover it actually works great, so just try it!